astropy_healpix/_core.c

```diff
@@ -0,0 +1,368 @@
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+
+#include <Python.h>
+#include <numpy/arrayobject.h>
+#include <numpy/ufuncobject.h>
+#include "healpix.h"
+#include "healpix-utils.h"
+#include "interpolation.h"
+
+/* FIXME: Remove this once we drop Python 2 support. */
+#include "six.h"
+
+/* FIXME: We need npy_set_floatstatus_invalid(), but unlike most of the Numpy
+ * C API it is only available on some platforms if you explicitly link against
+ * Numpy, which is not typically done for building C extensions. This bundled
+ * header file contains a static definition of _npy_set_floatstatus_invalid().
+ */
+#include "ieee754.h"
+
+
+typedef struct {
+    int64_t (*order_to_xy)(int64_t, int);
+    int64_t (*xy_to_order)(int64_t, int);
+} order_funcs;
```

Comment:
Maybe leave a short comment what `order_funcs` is, and how it's used by the ufuncs via `data`?
For someone that's not familiar with ufuncs like me (and most Astropy devs as possible future maintainers), that would be helpful.

I was reading through  http://www.numpy.org/devdocs/user/c-info.ufunc-tutorial.html and it doesn't contain an explanation / example using `data`, so while I find the ufunc code and signature overall very cryptic, this was the part I found most cryptic.

Reply:
OK, I added a comment. Does it look OK?